Adopt more smart pointers in LocalFrame

rwlbuis opened the pull request:
#### 088a3e573b59664dc988291e52900bba6b106b2c
<pre>
Adopt more smart pointers in LocalFrame
<a href="https://bugs.webkit.org/show_bug.cgi?id=273397">https://bugs.webkit.org/show_bug.cgi?id=273397</a>

Reviewed by Sihui Liu.

Adopt more smart pointers in LocalFrame based on the
[alpha.webkit.UncountedLocalVarsChecker] warning.

* Source/WebCore/page/LocalFrame.cpp:
(WebCore::parentPageZoomFactor):
(WebCore::parentTextZoomFactor):
(WebCore::rootFrame):
(WebCore::LocalFrame::LocalFrame):
(WebCore::LocalFrame::~LocalFrame):
(WebCore::LocalFrame::shouldUsePrintingLayout const):
(WebCore::LocalFrame::frameForWidget):
(WebCore::LocalFrame::rangeForPoint):
(WebCore::LocalFrame::debugDescription const):
(WebCore::LocalFrame::contentFrameFromWindowOrFrameElement):

Canonical link: <a href="https://commits.webkit.org/278206@main">https://commits.webkit.org/278206@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/881201e9bc024a5aee86584544e1a1ca302f3c48

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/49745 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/29032 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/52795 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/52986 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/420 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/52049 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/35053 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/26649 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/40602 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/51844 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/26567 "Passed tests") | [❌ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/42824 "Found 1 new API test failure: TestWebKitAPI.ResourceLoadStatistics.GrandfatherCallback (failure)") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/21705 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/24007 "Passed tests") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/44031 "Build is being retried. Recent messages:OS: Monterey (12.6.8), Xcode: 13.4.1; Skipping applying patch since patch_id isn't provided; Checked out pull request; Skipped layout-tests; layout-tests running") | [✅ 🛠 wpe-skia](https://ews-build.webkit.org/#/builders/52/builds/8116 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/45940 "Passed tests") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/44546 "Build is being retried. Recent messages:OS: Monterey (12.6.8), Xcode: 13.4.1; Skipping applying patch since patch_id isn't provided; Checked out pull request") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/54568 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/24832 "Built successfully") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/20989 "Build is being retried. Recent messages:OS: Sonoma (14.4.1), Xcode: 15.3; Skipping applying patch since patch_id isn't provided; Checked out pull request; Skipped layout-tests; 5 flakes 1 failures; Uploaded test results; layout-tests running") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/47962 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/26098 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/42945 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/47006 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/10930 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/26946 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/25823 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

cdumez commented on Source/WebCore/page/LocalFrame.cpp:
```diff
@@ -134,23 +134,23 @@ DEFINE_DEBUG_ONLY_GLOBAL(WTF::RefCountedLeakCounter, frameCounter, ("Frame"));
+    RefPtr parent = dynamicDowncast<LocalFrame>(frame->tree().parent());
     if (!parent)
         return 1;
     return parent->pageZoomFactor();
```
This looks like a trivial inline function, no? So why are we doing smart pointer adoption here? Please follow the guidelines.

rwlbuis replied:
Sorry, indeed I did not check for trivial inline functions. Fixed this one now and went over all other changes to check for this.

szewai commented on Source/WebCore/page/LocalFrame.cpp:
```diff
@@ -1217,9 +1219,9 @@ LocalFrame* LocalFrame::contentFrameFromWindowOrFrameElement(JSContextRef contex
+    Ref vm = globalObject->vm();
 
-    if (auto* window = JSDOMWindow::toWrapped(vm, value))
+    if (RefPtr window = JSDOMWindow::toWrapped(vm, value))
```
It seems the local variable vm is unnecessary

rwlbuis replied:
Removed the local variable.

rwlbuis committed:
Adopt more smart pointers in LocalFrame
https://bugs.webkit.org/show_bug.cgi?id=273397

Reviewed by Sihui Liu.

Adopt more smart pointers in LocalFrame based on the
[alpha.webkit.UncountedLocalVarsChecker] warning.

* Source/WebCore/page/LocalFrame.cpp:
(WebCore::parentPageZoomFactor):
(WebCore::parentTextZoomFactor):
(WebCore::rootFrame):
(WebCore::LocalFrame::LocalFrame):
(WebCore::LocalFrame::~LocalFrame):
(WebCore::LocalFrame::shouldUsePrintingLayout const):
(WebCore::LocalFrame::frameForWidget):
(WebCore::LocalFrame::rangeForPoint):
(WebCore::LocalFrame::debugDescription const):
(WebCore::LocalFrame::contentFrameFromWindowOrFrameElement):

Canonical link: https://commits.webkit.org/278206@main
